tools/src/main/java/org/apache/kafka/tools/FeatureCommand.java

```diff
@@ -0,0 +1,331 @@
+        Map<String, FeatureUpdate> updates = new HashMap<>();
+        MetadataVersion version;
+        String metadata = namespace.getString("metadata");
+        if (metadata !=  null) {
+            try {
+                version = MetadataVersion.fromVersionString(metadata);
+            } catch (Throwable e) {
+                throw new TerseException("Unsupported metadata version " + metadata +
+                        ". Supported metadata versions are " + metadataVersionsToString(
+                        MetadataVersion.MINIMUM_BOOTSTRAP_VERSION, MetadataVersion.latest()));
+            }
+            updates.put(MetadataVersion.FEATURE_NAME, new FeatureUpdate(version.featureLevel(), upgradeType));
+        }
+
+        List<String> features = namespace.getList("feature");
+        if (features != null) {
+            features.forEach(feature -> {
+                String[] nameAndLevel;
+                nameAndLevel = parseNameAndLevel(feature);
+
+                if (updates.put(nameAndLevel[0], new FeatureUpdate(Short.parseShort(nameAndLevel[1]), upgradeType)) != null) {
+                    throw new RuntimeException("Feature " + nameAndLevel[0] + " was specified more than once.");
+                }
+            });
+        }
+
+        update(op, admin, updates, namespace.getBoolean("dry_run"));
+    }
+
+    static void handleDisable(Namespace namespace, Admin adminClient) throws TerseException {
+        FeatureUpdate.UpgradeType upgradeType = downgradeType(namespace);
+        Map<String, FeatureUpdate> updates = new HashMap<>();
+
+        namespace.getList("feature").forEach(feature -> {
+            if (updates.put((String) feature, new FeatureUpdate((short) 0, upgradeType)) != null) {
+                throw new RuntimeException("Feature " + feature + " was specified more than once.");
+            }
+        });
+
+        update("disable", adminClient, updates, namespace.getBoolean("dry_run"));
+    }
+
+    private static void update(String op, Admin admin, Map<String, FeatureUpdate> updates, Boolean dryRun) throws TerseException {
+        if (updates.isEmpty()) {
+            throw new TerseException("You must specify at least one feature to " + op);
+        }
+
+        UpdateFeaturesResult result = admin.updateFeatures(updates, new UpdateFeaturesOptions().validateOnly(dryRun));
+        Map<String, Optional<Throwable>> errors = new HashMap<>();
+        result.values().forEach((feature, future) -> {
+            try {
+                future.get();
+                errors.put(feature, null);
+            } catch (ExecutionException e) {
+                errors.put(feature, Optional.ofNullable(e.getCause()));
+            } catch (Throwable t) {
+                errors.put(feature, Optional.of(t));
+            }
+        });
+
+        AtomicInteger numFailures = new AtomicInteger();
+        errors.keySet().forEach(feature -> {
```

Comment:
I don't think we need to introduce the `AtomicInteger` just for `forEach`. Could we use traditional for loop instead here directly?